src/batou/lib/file.py

```diff
@@ -19,18 +20,31 @@
 
 
 def ensure_path_nonexistent(path):
+    """Ensure that the given path does not exist.
+
+    strategy: rename the file to a temporary name, then remove it.
+    this allows us to fail in a useful way, since rename is atomic.
+    If we detect that the file exists after the rename, we know that
+    something else is going on and we can bail out.
+    """
+    tmp_nonce = random.getrandbits(32)
+    tmp_filename = path + ".batou-ensure-path-nonexistent-tmp" + str(tmp_nonce)
```

Comment:
Here's a suggestion to make this less error prone and remove the race condition on the temporary directory:

a) use the tempfile module
b) create a temporary directory (in the right location!) via the module, which is guaranteed to succeed (or the disk is full or there are weird permisison problems)
c) rename the original *into* that directory so you do not get any further race conditions
d) recursively delete the temporary directory, that means we can now just simply rmtree in all cases, that's nice, too!